docs/docsite/rst/network/dev_guide/developing_plugins_network.rst

```diff
@@ -0,0 +1,108 @@
+
+.. _developing_modules_network:
+.. _developing_plugins_network:
+
+**************************
+Network connection plugins
+**************************
+
+Each network connection plugin has a set of plugins which provide a specification of the connection for a particular set of devices. Public methods of these plugins may be called on the connection proxy object from the module just as connection methods can.
+
+.. code-block:: python
+
+  from ansible.module_utils.connection import Connection
+
+  ...
+
+  # module is your AnsibleModule instance
+  connection = Connection(module._socket_path)
+
+  # You can now call any method (that doesn't start with '_') of the connection
+  # plugin or its platform-specific plugin
+  connection.get_config()
+
+.. contents::
+   :local:
+
+.. _developing_plugins_httpapi:
+
+Developing httpapi plugins
+==========================
+
+:ref:`httpapi plugins <httpapi_plugins>` serve as adapters for various HTTP(S) APIs for use with the ``httpapi`` connection plugin. They should implement a minimal set of convenience methods tailored to the API you are attempting to use.
+
+Specifically, there are a few methods that the ``httpapi`` connection plugin expects to exist.
+
+Making requests
+---------------
+
+The ``httpapi`` connection plugin has a ``send()`` method, but an httpapi plugin needs a ``send_request(self, data, **message_kwargs)`` method as a higher-level wrapper to ``send()``. This method should prepare requests by adding fixed values like common headers or URL root paths. This method may do more complex work such as turning data into formatted payloads, or determining which path or method to request. It may then also unpack responses to be more easily consumed by the caller.
+
+.. code-block:: python
+
+   def send_request(self, data, path, method='POST'):
+       # Fixed headers for requests
+       headers = {'Content-Type': 'application/json'}
+       try:
+           response, response_content = self.connection.send(path, data, method=method, headers=headers)
```

Comment:
This particular example uses a `POST` method so it will contain body, probably represented by data. What happens with a `GET` request?

Reply:
This example defaults to `POST`, but could just as easily be called with `GET` (or `PATCH`, or `PUT`, or `DELETE`), The point is that this method should take whatever information it needs from a module or module_utils and turns that into a valid request for the API it follows. That could involve packing `data` into a JSON payload, or ignoring it if it sees that `method` is `'GET'`, or even just passing it through as shown and trusting whatever calls it to give it the right format. What happens in any case is up to the needs of your plugin, which is why it isn't implemented in the base class.

Reply:
So this could be a valid implementation? I'm just swagging it, not sure it works...
```
def send_request(self, data, path, method='POST'):
    if method == 'POST':
        response, response_content = self.connection.send(path, data, method=method, headers=headers)
    elif method == 'GET':
        response, response_content = self.connection.send(path, None, method=method, headers=headers)
```

Reply:
I would probably do

```
if method == 'GET':
    data = None
response, response_content = self.connection.send(path, data, method=method, headers=headers)
```

Unless you had some other reason to separate the `send()` calls, but that would work as well.